src/shared/particle_neighborhood/neighborhood.h

```diff
@@ -74,7 +75,7 @@ using ParticleConfiguration = StdLargeVec<Neighborhood>;
 class NeighborhoodDevice {
   public:
     size_t* current_size_;	/**< the current number of neighbors */
-    static constexpr size_t allocated_size_ = 40;  /**< the limit of neighbors does not require memory allocation  */
+    static constexpr size_t allocated_size_ = 28;  /**< the limit of neighbors does not require memory allocation  */
```

Comment:
Why do this change?

Reply:
New memory cannot be allocated within kernel execution, for this reason an upper bound of neighbors needs to be defined. I tested that, in the 2d case, neighbors are around 26, hence 40 was giving too much slack and wasted device memory. 28 is a better upper bound.

To be fair, those values don't need to be constexpr, they can also be computed at runtime when the neighborhood is first defined.

Since the maximum number of neighbors depends on the smoothing length, a better approach would be to implement a function returning the upper bound based on the currently used smoothing kernel. But I have not checked how coupled the smoothing kernel and the neighborhood construction are.